lib/bounce_email.rb

```diff
@@ -237,5 +239,38 @@ def extract_original_message_after_delimiter(mail, delimiter_index)
       message = mail.body.to_s.split(delimiter).last
       message.split(INLINE_MESSAGE_END_DELIMITER).first.strip if message.match(INLINE_MESSAGE_END_DELIMITER)
     end
+
+    def extract_and_assign_fields_from_original_mail(mail)
+      extract_and_assign_message_id(mail) if mail.message_id.nil?
+      extract_and_assign_from(mail) if mail.from.nil?
+      extract_and_assign_to(mail) if mail.to.nil?
+      extract_and_assign_subject(mail) if mail.subject.nil?
+      mail
+    end
+
+    def extract_and_assign_message_id(mail)
```

Comment:
I'm a fan of small methods as they're easier to test. However, since this is a private method testing is sort of irrelevant. Your suggestion certainly solves the naming dilemma. I'll implement it.